WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/table/WNPRC_PurchasingCustomizer.java

```diff
@@ -38,16 +38,13 @@ else if (matches(tableInfo, "ehr_purchasing", "purchasingReceiverOverview"))
 
     private void addRequestLink(AbstractTableInfo ti)
     {
-        if (ti.hasPermission(Objects.requireNonNull(ti.getUserSchema()).getUser(), AdminPermission.class))
-        {
-            ActionURL returnUrl = new ActionURL(WNPRC_PurchasingController.PurchaseAdminAction.class, ti.getUserSchema().getContainer());
-            ActionURL detailsUrl = new ActionURL(WNPRC_PurchasingController.PurchasingRequestAction.class, ti.getUserSchema().getContainer());
-            detailsUrl.addParameter("requestRowId", "${rowId}");
-            detailsUrl.addParameter("returnUrl", returnUrl.getPath());
+        ActionURL returnUrl = new ActionURL(WNPRC_PurchasingController.PurchaseAdminAction.class, ti.getUserSchema().getContainer());
+        ActionURL detailsUrl = new ActionURL(WNPRC_PurchasingController.PurchasingRequestAction.class, ti.getUserSchema().getContainer());
+        detailsUrl.addParameter("requestRowId", "${rowId}");
+        detailsUrl.addParameter("returnUrl", returnUrl.getPath());
```

Comment:
I think you want toString() instead of getPath() here. getPath() won't include GET parameters, if any.

Reply:
Done! 
Also, realized return url was all set to go the Purchase Admin page for different user types, so i've adjusted the returnURL based on user perm.